src/renderer/AsyncResourceGatherer.cpp

```diff
@@ -42,7 +42,8 @@ CAsyncResourceGatherer::CAsyncResourceGatherer() {
     }
 
     for (auto& mon : mons) {
-        const auto MON = std::find_if(g_pHyprlock->m_vOutputs.begin(), g_pHyprlock->m_vOutputs.end(), [mon](const auto& other) { return other->stringPort == mon; });
+        const auto MON = std::find_if(g_pHyprlock->m_vOutputs.begin(), g_pHyprlock->m_vOutputs.end(),
+                                      [mon](const auto& other) { return other->stringPort == mon || other->description.find(mon) == 0; });
```

Comment:
you have to check for empty desc

also how the fuck are you comparing a pointer to a string???

Reply:
ehm, I didn't even change this.
isn't `mon` a member of vector of strings? (from widget conf) [here](https://github.com/hyprwm/hyprlock/pull/216/files#diff-7cc329661269fc3fc040b387dac691ae82028fce0622b565340fc08274047c33R24)
![205854-area- 971 607-634x179](https://github.com/hyprwm/hyprlock/assets/130279855/d9b8e490-bde4-440c-96d3-a2070ba1c8cc)

Reply:
or do you mean description field? it's string [here](https://github.com/hyprwm/hyprlock/pull/216/files#diff-588e4c1eeffc3faee9f156543179767a6a4c8f352637af7d3e0bfb7d7484185bR50)
should've named it different

Reply:
no I mean `mon`

Reply:
the same like it is and was compared to `stringPort`? I didn't even touch this part. Or I can't get you.
isn't `mon` a member of vector of strings (from widgets conf) [here](https://github.com/hyprwm/hyprlock/pull/216/files#diff-7cc329661269fc3fc040b387dac691ae82028fce0622b565340fc08274047c33R24)?

![184107-area- 117 371-904x176](https://github.com/hyprwm/hyprlock/assets/130279855/1820c6f2-f531-4752-b571-4072f36cd437)

![184250-area- 719 447-766x182](https://github.com/hyprwm/hyprlock/assets/130279855/8bb35c6f-bdf9-4c92-85f4-888aecee025d)

Reply:
oh god am dum